### PR TITLE
Remove Space component from header and footer

### PR DESCRIPTION
**Pull request by alebedev**
## Describe your changes

Replace `Space` component with gap styles in header and footer

## Justify why they are needed

I hate `<Space>` in our code - it targets child elements in non-obvious and computationally expensive way. There's also a risk of layout shift if any of its children add inline style tag

gap / rowGap / columnGap is simpler idiomatic way to get the same effect

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by alebedev** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4390?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#4390** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4390?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#4389** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4389?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4387** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4387?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @alebedev and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/store/src/components/Header/Header.css.ts [hunk 1]
```diff
@@ -198,13 +198,15 @@ export const navigationContent = style({
...
   paddingBottom: tokens.space.xl,
+  rowGap: tokens.space.lg,
 
   '@media': {
     [minWidth.lg]: {
       backgroundColor: tokens.colors.light,
       boxShadow: tokens.shadow.default,
       borderRadius: tokens.radius.sm,
       padding: tokens.space.md,
+      gap: tokens.space.md,
```
Comment (gustaveen) [inline apps/store/src/components/Header/Header.css.ts:209]: Shouldn't we just set `rowGap` here?

**Review by gustaveen — APPROVED** [review]
Looks good, just wondering about one property

**Comment by alebedev** [comment]
### Merge activity

* **Jun 20, 4:40 AM EDT**: [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4390) rebased this pull request after merging its parent, because this pull request is set to merge when ready.

**Pull request merged**